cwarhm/data_specific_processing/era5.py

```diff
@@ -0,0 +1,227 @@
+            variables_pres_convert = ['t' ,'q']
+            attr_names_expected = ['scale_factor' ,'add_offset' ,'_FillValue' ,'missing_value' ,'units' ,'long_name'
+                                   ,'standard_name'] # these are the attributes we think each .nc variable has
+            loop_attr_copy_these = ['units' ,'long_name'
+                                    ,'standard_name'] # we will define new values for _FillValue and missing_value when writing the .nc variables' attributes
+
+            # Open the destination file and transfer information
+            with nc4.Dataset(forcingPath / data_pres) as src1, nc4.Dataset(forcingPath / data_surf) as src2, nc4.Dataset \
+                    (mergePath / data_dest, "w") as dest:
+
+                # === Some general attributes
+                dest.setncattr('History' ,'Created ' + time.ctime(time.time()))
+                dest.setncattr('Language' ,'Written using Python')
+                dest.setncattr('Reason'
+                               ,'(1) ERA5 surface and pressure files need to be combined into a single file (2) Wind speed U and V components need to be combined into a single vector (3) Forcing variables need to be given to SUMMA without scale and offset')
+
+                # === Meta attributes from both sources
+                for name in src1.ncattrs():
+                    dest.setncattr(name + ' (pressure level (10m) data)', src1.getncattr(name))
+                for name in src2.ncattrs():
+                    dest.setncattr(name + ' (surface level data)', src1.getncattr(name))
+
+                # === Dimensions: latitude, longitude, time
+                # NOTE: we can use the lat/lon from the surface file (src2), because those are already in proper units. If there is a mismatch between surface and pressure we shouldn't have reached this point at all due to the check above
+                for name, dimension in src2.dimensions.items():
+                    if dimension.isunlimited():
+                        dest.createDimension( name, None)
+                    else:
+                        dest.createDimension( name, len(dimension))
+
+                # === Get the surface level generic variables (lat, lon, time)
+                for name, variable in src2.variables.items():
+
+                    # Transfer lat, long and time variables because these don't have scaling factors
+                    if name in variables_surf_transfer:
+                        dest.createVariable(name, variable.datatype, variable.dimensions, fill_value = -999)
+                        dest[name].setncatts(src1[name].__dict__)
+                        dest.variables[name][:] = src2.variables[name][:]
+
+                # === For the forcing variables, we need to:
+                # 1. Extract them (this automatically applies scaling and offset with nc4) and apply non-negativity constraints
+                # 2. Create a .nc variable with the right SUMMA name and file type
```

Comment:
merging and renaming are two different actions - ideally these are separate functions. But not part of the scope now I would say.

Reply:
Agree